toolkit/tools/pkg/imagecustomizerlib/artifactsinputoutput.go

```diff
@@ -111,3 +157,145 @@ func outputArtifacts(items []imagecustomizerapi.OutputArtifactsItemType, outputD
 
 	return nil
 }
+
+func replaceSuffix(input string, oldSuffix string, newSuffix string) string {
+	if !strings.HasSuffix(input, oldSuffix) {
+		return input
+	}
+	return strings.TrimSuffix(input, oldSuffix) + newSuffix
+}
+
+func writeInjectFilesYaml(metadata []imagecustomizerapi.InjectArtifactMetadata, outputDir string) error {
+	yamlStruct := imagecustomizerapi.InjectFilesConfig{
+		InjectFiles:     metadata,
+		PreviewFeatures: []imagecustomizerapi.PreviewFeature{imagecustomizerapi.PreviewFeatureInjectFiles},
+	}
+
+	yamlBytes, err := yaml.Marshal(&yamlStruct)
+	if err != nil {
+		return fmt.Errorf("failed to marshal inject files metadata: %w", err)
+	}
+
+	outputFilePath := filepath.Join(outputDir, "inject-files.yaml")
+	if err := os.WriteFile(outputFilePath, yamlBytes, 0o644); err != nil {
+		return fmt.Errorf("failed to write inject-files.yaml: %w", err)
+	}
+
+	return nil
+}
+
+func InjectFilesWithConfigFile(buildDir string, configFile string, inputImageFile string,
+	outputImageFile string, outputImageFormat string,
+) error {
+	var injectConfig imagecustomizerapi.InjectFilesConfig
+	err := imagecustomizerapi.UnmarshalYamlFile(configFile, &injectConfig)
+	if err != nil {
+		return err
+	}
+
+	if err := injectConfig.IsValid(); err != nil {
+		return fmt.Errorf("invalid inject files config: %w", err)
+	}
+
+	baseConfigPath, _ := filepath.Split(configFile)
+
+	absBaseConfigPath, err := filepath.Abs(baseConfigPath)
+	if err != nil {
+		return fmt.Errorf("failed to get absolute path of inject-files.yaml:\n%w", err)
+	}
+
+	err = InjectFiles(buildDir, absBaseConfigPath, inputImageFile, injectConfig.InjectFiles,
+		outputImageFile, outputImageFormat)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func InjectFiles(buildDir string, baseConfigPath string, inputImageFile string,
```

Comment:
Please add some functional tests for both inject files and output files.

Reply:
WIP..